From e0077ab2c2d964fa00f1b83b93b6edec0667fe36 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 28 Nov 2017 17:02:22 +0800
Subject: [PATCH 2946/5242] MLK-17034-3: ASoC: imx-wm8960: remove clk
 operation in startup/shutdown

commit  ba8e48f00d400b8e743b945e17904ed76c634b49 from
https://source.codeaurora.org/external/imx/linux-imx.git

In imx8 when systerm enter suspend state, the power of subsystem will be
off, the clock enable state will be lost after resume, the startup
function isn't called after resume, so the clock will be enabled after
resume, the clock operation should be moved to pm runtime resume function.
For the mclk is for codec, this clock enablement and disablement will be
move to code driver's runtime resume and runtime suspend

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8960.c |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index b905f2c..5065e5e 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -313,12 +313,6 @@ static int imx_hifi_startup(struct snd_pcm_substream *substream)
 			return ret;
 	}
 
-	ret = clk_prepare_enable(data->codec_clk);
-	if (ret) {
-		dev_err(card->dev, "Failed to enable MCLK: %d\n", ret);
-		return ret;
-	}
-
 	return ret;
 }
 
@@ -329,8 +323,6 @@ static void imx_hifi_shutdown(struct snd_pcm_substream *substream)
 	struct imx_wm8960_data *data = snd_soc_card_get_drvdata(card);
 	bool tx = substream->stream == SNDRV_PCM_STREAM_PLAYBACK;
 
-	clk_disable_unprepare(data->codec_clk);
-
 	data->is_stream_opened[tx] = false;
 }
 
-- 
1.7.9.5

